packages/cms-lib/lib/config.js

```diff
@@ -353,6 +377,118 @@ const deleteEmptyConfigFile = () => {
   );
 };
 
+const getConfigVariablesFromEnv = () => {
+  const env = process.env;
+
+  return {
+    apiKey: env[HUBSPOT_API_KEY],
+    clientId: env[HUBSPOT_CLIENT_ID],
+    clientSecret: env[HUBSPOT_CLIENT_SECRET],
+    personalAccessKey: env[HUBSPOT_PERSONAL_ACCESS_KEY],
+    portalId: parseInt(env[HUBSPOT_PORTAL_ID], 10),
+    refreshToken: env[HUBSPOT_REFRESH_TOKEN],
+  };
+};
+
+const generatePersonalAccessKeyPortalConfig = (portalId, personalAccessKey) => {
+  return {
+    defaultPortal: ENVIRONMENT_VARIABLES_DEFAULT_PORTAL_NAME,
+    portals: [
+      {
+        inMemoryOnly: true,
+        name: ENVIRONMENT_VARIABLES_DEFAULT_PORTAL_NAME,
+        authType: PERSONAL_ACCESS_KEY_AUTH_METHOD.value,
+        portalId,
+        personalAccessKey,
+      },
+    ],
+  };
+};
+
+const generateOauthPortalConfig = (
+  portalId,
+  clientId,
+  clientSecret,
+  refreshToken,
+  scopes
+) => {
+  return {
+    defaultPortal: ENVIRONMENT_VARIABLES_DEFAULT_PORTAL_NAME,
+    portals: [
+      {
+        inMemoryOnly: true,
+        name: ENVIRONMENT_VARIABLES_DEFAULT_PORTAL_NAME,
+        authType: OAUTH_AUTH_METHOD.value,
+        portalId,
+        auth: {
+          clientId,
+          clientSecret,
+          scopes,
+          tokenInfo: {
+            refreshToken,
+          },
+        },
+      },
+    ],
+  };
+};
+
+const generateApiKeyPortalConfig = (portalId, apiKey) => {
+  return {
+    defaultPortal: ENVIRONMENT_VARIABLES_DEFAULT_PORTAL_NAME,
+    portals: [
+      {
+        inMemoryOnly: true,
```

Comment:
Is there a reason that this is being handled in the config object instead of elsewhere? I think we want to create a clear separation between the interface of the config data and other data?

Reply:
The handling of this is now done through setting `disableWritesToFile` to `true` when we are using an environment variable based config